book/B-embedding-git/sections/dulwich.asc

```diff
@@ -0,0 +1,44 @@
+=== Dulwich
+
+(((Dulwich)))((("Python")))
+There is also a pure-Python Git implementation - Dulwich.
+The project is hosted under https://www.dulwich.io/
+It aims to provide an interface to git repositories (both local and remote) that doesn't call out to git directly but instead uses pure Python.
+It has an optional C extensions though, that significantly improve the performance.
+
+Dulwich follows git design and separate two basic levels of API: plumbing and porcelain.
+
+Here is an example of using the lower level API to access the commit message of the last commit:
+
+[source, python]
+-----
+from dulwich.repo import Repo
+r = Repo('.')
```

Comment:
Would you mind renaming this variable to `current_repo` or something similar? And maybe `c` should be called `commit`.

Reply:
Right now it's just 1-to-1 copy from https://www.dulwich.io/docs/#getting-started

Please, let me know if you still think it should be change and I will be happy to do so.

Reply:
Hmm. Should we make this example more readable, or easier to update? Let's stick with Dulwich's example then.